File: src/models/notebook-punishment.ts
import type { Attachment } from "./attachment";

export type NotebookPunishment = Readonly<{
  id: string;
  title: string;
  reasons: string[];

  isDuringLesson: boolean;
  exclusion: boolean;

  workToDo: string;
  workToDoDocuments: Array<Attachment>;

  circumstances: string;
  circumstancesDocuments: Array<Attachment>;

  giver: string;
  dateGiven: Date;

  durationMinutes: number;
}>;

File: library/src/commonMain/kotlin/models/NotebookPunishment.kt
package models

import kotlinx.datetime.LocalDateTime

data class NotebookPunishment(
    val id: String,
    val title: String,
    val reasons: List<String>,

    val isDuringLesson: Boolean,
    val exclusion: Boolean,

    val workToDo: String,
    val workToDoDocuments: List<Attachment>,

    val circumstances: String,
    val circumstancesDocuments: List<Attachment>,

    val giver: String,
    val dateGiven: LocalDateTime,

    val durationMinutes: Int
)